src/commons/api/rrdp.rs

```diff
@@ -313,6 +316,29 @@ impl CurrentObjects {
         }
     }
 
+    /// Returns a copy of self where elements matching the given URI
+    /// are removed if there are any matches. Otherwise, returns None.
+    pub fn purge(&self, uri: &uri::Rsync) -> Option<Self> {
```

Comment:
This fn doesn't actually purge anything, it doesn't even change `self`... it could even always return a collection in place of Option and just be empty if there is nothing to "purge", maybe it should be called find matching or find purgeable or ...

Reply:
ack - changed it to `with_matching_uri_deleted`